### PR TITLE
Add support to Cucumber Expressions

### PR DESCRIPTION
**Pull request by Dbuggerx**
In this PR I've:

* Updated the dependencies;
* Added support to [Cucumber Expressions](https://cucumber.io/docs/cucumber/cucumber-expressions/);
* Improved documentation and examples.

I'd also like to thank everyone involved in this project for such good work!

### CODE REVIEW

**Comment by Dbuggerx** [comment]
Hey @Lukas-Kullmann , I've done some changes. Please let me know if it's ok now. Thanks.

File: README.md [hunk 1]
```diff
@@ -174,6 +186,55 @@ Examples:
...
+
+Besides using Regular Expressions, you can also use [Cucumber Expressions](https://cucumber.io/docs/cucumber/cucumber-expressions/) in your steps, and have support for [Optional text](https://cucumber.io/docs/cucumber/cucumber-expressions/#optional-text), [Alternative text](https://cucumber.io/docs/cucumber/cucumber-expressions/#alternative-text) and getting parameters in their desired types. The [Cucumber built-in parameter types](https://cucumber.io/docs/cucumber/cucumber-expressions/#parameter-types) are supported by default.
+
+It's also possible to add custom parameter types by creating a file that exports a `cucumberExpressions.ParameterTypeRegistry`, and passing this file's path to the CLI with `--param-type-registry-file` or to the Programming interface with the `parameterTypeRegistryFile` method.
+
+Example:
+
+1. Create a _ParameterTypeRegistry_ (e.g. _myCustomParamRegistry.js_):
+
+        const { ParameterTypeRegistry, ParameterType } = require('cucumber-expressions');
```
Comment (Lukas-Kullmann) [inline README.md]: Please add language hints to the code blocks:

```suggestion
    ```js
    const { ParameterTypeRegistry, ParameterType } = require('cucumber-expressions');
```

Please also do this for the other code blocks in the readme file.

**Comment by Lukas-Kullmann** [comment]
Hey @Dbuggerx,

thank you for you contribution. I published v2.3.0 with your changes.

**Comment by Dbuggerx** [comment]
Thank you @Lukas-Kullmann !

**Pull request merged**